cmake/scripts/buildinfo.cmake

```
@@ -19,20 +19,19 @@ if (NOT ETH_BUILD_PLATFORM)
 	set(ETH_BUILD_PLATFORM "unknown")
 endif()
 
-# Logic here: If prerelease.txt exists but is empty, it is a non-pre release.
-# If it does not exist, create our own prerelease string
-if (EXISTS ${ETH_SOURCE_DIR}/prerelease.txt)
-	file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
-	string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
-else()
+# Logic here: If prerelease.txt is empty, or if it has the templated placeholder, it is a non-pre release.
+file(READ ${ETH_SOURCE_DIR}/prerelease.txt SOL_VERSION_PRERELEASE)
+string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
+If (SOL_VERSION_PRERELEASE MATCHES [$][F][o][r][m][a][t][:])
```

Also, It would probably be best to require the match to be at the beginning of the line to avoid weird false-positives:
```suggestion
If (SOL_VERSION_PRERELEASE MATCHES ^[$][F][o][r][m][a][t][:])
```